src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,399 @@
+    (*depthTexture)
+        .setWrapping(Mn::GL::SamplerWrapping::ClampToEdge)
+        .setMinificationFilter(Mn::GL::SamplerFilter::Nearest)
+        .setMagnificationFilter(Mn::GL::SamplerFilter::Nearest)
+        .setStorage(1, Mn::GL::TextureFormat::DepthComponent32F, size);
+  }
+}
+
+void CubeMap::recreateFramebuffer() {
+  Mn::Vector2i viewportSize{imageSize_, imageSize_};
+  frameBuffer_ = Mn::GL::Framebuffer{{{}, viewportSize}};
+  optionalDepthBuffer_.setStorage(Mn::GL::RenderbufferFormat::DepthComponent24,
+                                  viewportSize);
+}
+
+Mn::GL::CubeMapCoordinate CubeMap::convertFaceIndexToCubeMapCoordinate(
+    int faceIndex) {
+  CORRADE_ASSERT(faceIndex >= 0 && faceIndex < 6,
+                 "CubeMap::convertFaceIndexToCubeMapCoordinate(): the index of "
+                 "the cube side"
+                     << faceIndex << "is illegal.",
+                 Mn::GL::CubeMapCoordinate::PositiveX);
+  return Mn::GL::CubeMapCoordinate(int(Mn::GL::CubeMapCoordinate::PositiveX) +
+                                   faceIndex);
+}
+
+void CubeMap::prepareToDraw(int cubeSideIndex) {
+  Magnum::GL::CubeMapCoordinate cubeMapCoord =
+      convertFaceIndexToCubeMapCoordinate(cubeSideIndex);
+
+  if (flags_ & Flag::ColorTexture) {
+    frameBuffer_.attachCubeMapTexture(
+        colorAttachment, *textures_[TextureType::Color], cubeMapCoord, 0);
+  }
+
+  if (flags_ & Flag::DepthTexture) {
+    frameBuffer_.attachCubeMapTexture(
+        Mn::GL::Framebuffer::BufferAttachment::Depth,
+        *textures_[TextureType::Depth], cubeMapCoord, 0);
+  } else {
+    frameBuffer_.attachRenderbuffer(
+        Mn::GL::Framebuffer::BufferAttachment::Depth, optionalDepthBuffer_);
+  }
+
+  mapForDraw();
+
+  frameBuffer_.clearDepth(1.0f).clearColor(1, Mn::Vector4ui{0});
+
+  CORRADE_INTERNAL_ASSERT(
+      frameBuffer_.checkStatus(Mn::GL::FramebufferTarget::Draw) ==
+      Mn::GL::Framebuffer::Status::Complete);
+}
+
+void CubeMap::mapForDraw() {
+  frameBuffer_.mapForDraw({
+      {Mn::Shaders::Generic3D::ColorOutput, colorAttachment},
+      // TODO:
+      //{Mn::Shaders::Generic3D::ObjectIdOutput, objectIdAttachment}
+  });
+}
+void CubeMap::textureTypeSanityCheck(TextureType type,
+                                     const std::string& functionNameStr) {
+  switch (type) {
+    case TextureType::Color:
+      CORRADE_ASSERT(flags_ & Flag::ColorTexture,
+                     functionNameStr.c_str()
+                         << "instance was not created with color "
+                            "texture output enabled.", );
+      break;
+    case TextureType::Depth:
+      CORRADE_ASSERT(flags_ & Flag::DepthTexture,
+                     functionNameStr.c_str()
+                         << "instance was not created with depth "
+                            "texture output enabled.", );
+      break;
+    default:
+      CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+      break;
+  }
```

Comment:
> But what if a developer ignores that warning?

Sure. The point is that the developers *can* get the warning at build time already if they choose to pay attention to new warnings, instead of later at runtime (where the code path might not even get hit until several months later).